File: code/matlab/lab6.m
clc;
clear all;
close all;

fc = 900*10^6;                                      % Carrier Frequency, fc
wavelength = (3*10^8)/(900*10^6);    % Wavelength
v = 70*(1000/(60*60));                        % Vehicle speed, v

%  Solution (a) 
The_received_frequency_of_A_is = (floor(fc)+(v/wavelength))/10^6;
fprintf('a) The received frequency of A is: %.6f MHz\n\n',The_received_frequency_of_A_is);

% Solution (b)
The_received_frequency_of_B_is = (floor(fc)-(v/wavelength))/10^6;
fprintf('b) The received frequency of B is: %f MHz\n\n',The_received_frequency_of_B_is);

% Solution (c)
disp('c) The received signal frequency is the same of the transmitted frequency 900 MHz');

File: code/python/lab6.py
Carrier_fre = 900*1000000
Velocity_of_light = 3*100000000
Wavelength = round((Velocity_of_light / Carrier_fre), 2)
Vehicle_speed = round((70000/3600), 2)
Fd = round((Vehicle_speed / Wavelength), 2)
Fa = (Carrier_fre + Fd)/1000000
Fb = (Carrier_fre - Fd)/1000000
print("The received carrier frequency for 'a':\n", Fa, 'MHz')
print("The received carrier frequency for 'b':\n", Fb, 'MHz')
print("The received carrier frequency for 'c':  \nSince the vehicle is moving perpendicular to the angle of arrival to the transmitted signal, there is no doppler effect. \nSo, the received signal's frequency is the same as the transmitted signal's frequency.\nSo, the received carrier frequency for 'c': 900 MHz")